Assets/Plugins/Krearthur/GOPainter/Scripts/GOPainter.cs

```diff
@@ -454,36 +406,30 @@ protected void OnActivation()
 
             CleanObjectGroups(true);
 
-            if (!TryGetComponent(out AudioSource audioSource))
-            {
-                audio = gameObject.AddComponent<AudioSource>();
-            }
-            else
-            {
-                audio = audioSource;
-            }
+            if (enableSounds)
+                audio.Init(this, resources);
 
             paintCanvas = GameObject.Find(paintCanvasName);
             if (paintCanvas == null)
             {
                 paintCanvas = GameObject.CreatePrimitive(PrimitiveType.Cube);
 
-                paintCanvas.GetComponent<Renderer>().sharedMaterial = canvasMaterial;
+                paintCanvas.GetComponent<Renderer>().sharedMaterial = resources.canvasMaterial;
                 paintCanvas.GetComponent<Renderer>().enabled = !hideCanvas;
                 paintCanvas.GetComponent<Renderer>().shadowCastingMode = UnityEngine.Rendering.ShadowCastingMode.Off;
                 paintCanvas.AddComponent<PaintCanvas>();
                 paintCanvas.hideFlags = HideFlags.HideInHierarchy;
                 paintCanvas.name = paintCanvasName;
                 paintCanvas.layer = LayerMask.NameToLayer(canvasLayer);
-                paintCanvas.transform.SetParent(groupRoot.transform, true);    
+                paintCanvas.transform.SetParent(groupRoot.transform, true);
             }
             UpdatePaintCanvasSize();
 
             IgnoredObjectsToPick = new GameObject[] { paintCanvas };
 
             parkedPicked = null;
             selectedPrefab = null;
-            if (prefabId < 0) prefabId = 0;
+            prefabId = 0;
```

Comment:
Fixed a bug. The error occurred if the current `prefabId` value is higher than currently possible (`paintObjects.Length` < `prefabId`)